plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.hdfs.ConfigurationUtils;
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.iceberg.IcebergConfig;
+import io.trino.plugin.iceberg.catalog.TrinoCatalog;
+import io.trino.plugin.iceberg.catalog.TrinoCatalogFactory;
+import io.trino.spi.security.ConnectorIdentity;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.rest.RESTSessionCatalog;
+
+import javax.annotation.concurrent.GuardedBy;
+import javax.inject.Inject;
+
+import java.net.URI;
+
+import static java.util.Objects.requireNonNull;
+
+public class TrinoIcebergRestCatalogFactory
+        implements TrinoCatalogFactory
+{
+    private final CatalogName catalogName;
+    private final String trinoVersion;
+    private final URI serverUri;
+    private final SecurityProperties securityProperties;
+    private final boolean isUniqueTableLocation;
+
+    @GuardedBy("this")
+    private RESTSessionCatalog icebergCatalog;
```

Comment:
We don't have to address this now, but it's problematic that `RESTSessionCatalog` depends on `org.apache.hadoop.conf.Configuration`. We're trying to eliminate Hadoop as a dependency for Iceberg, so we'll need to remove the dependency in `RESTSessionCatalog` or implement our own REST client here.

Reply:
This is actually a soft dependency that gets removed due to type erasure.  This should work fine without hadoop dependencies.